components/scroller/HorizontalScroller.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useRef } from 'react';
+
+const HorizontalScroller = ({ items }: any) => {
+  const scrollerRef = useRef(null);
+
+  useEffect(() => {
+    const scroller: any = scrollerRef.current;
+    let startOffset = 0;
+
+    const autoScroll = () => {
+      if (startOffset && startOffset >= scroller.scrollWidth - scroller.clientWidth) {
+        scroller.scrollTo({ left: 0 });
+        startOffset = 0;
+      } else {
+        startOffset += 1;
+      }
+      scroller.scrollTo({
+        left: startOffset,
+      });
+    };
+
+    const interval = setInterval(autoScroll, 16);
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
```

Comment:
please move the `ref` and `useEffect` to a hook
- suggestion: `useHorizontalScroller`